### PR TITLE
fix displaying subdirectories when selecting filters

### PR DESCRIPTION
**Pull request by YenguiSeddik**
fix sub element count of children directories

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 1]
```diff
@@ -462,12 +462,26 @@ public boolean elementExists(UUID parentDirectoryUuid, String elementName, Strin
...
 
-    public List<ElementAttributes> getElements(List<UUID> ids, boolean strictMode) {
+    public List<ElementAttributes> getElements(List<UUID> ids, boolean strictMode, List<String> types) {
         List<DirectoryElementEntity> elementEntities = directoryElementRepository.findAllById(ids);
+
         if (strictMode && elementEntities.size() != ids.stream().distinct().count()) {
             throw new DirectoryException(NOT_FOUND);
         }
-        return elementEntities.stream().map(ElementAttributes::toElementAttributes).collect(Collectors.toList());
+        if (types != null && !types.isEmpty()) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:471]: !CollectionUtils.isEmpty()  ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/directory/server/DirectoryService.java]: done

**Review by FranckLecuyer — APPROVED** [review]

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java [hunk 2]
```diff
@@ -899,6 +899,13 @@ public void testGetElement() {
                 toElementAttributes(scriptAttributes.getElementUuid(), "newScript", FILTER, null, "user1", null, scriptAttributes.getCreationDate())
             ))
         );
+
+        List<ElementAttributes> result = getElements(List.of(FILTER_UUID, UUID.randomUUID()), "user1", false, List.of(FILTER), 200);
```
Comment (archendel) [inline src/test/java/org/gridsuite/directory/server/DirectoryTest.java:903]: add a directory to the list to check that it is returned by this method

**Review by archendel — APPROVED** [review]
improve slightly the test 

Test: OK
Code: OK

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/test/java/org/gridsuite/directory/server/DirectoryTest.java]: ok done

**Pull request merged**